### PR TITLE
Add acceptance tests for how provider handles `region` and `zone` arguments, remove old unit tests

### PR DESCRIPTION
**Pull request by SarahFrench**
This PR removes unit tests for how the provider handles `region` and `zone`. Instead there are acc tests that use the new datasources `google_provider_config_plugin_framework` and `google_provider_config_sdk` 

**Release Note Template for Downstream PRs (will be copied)**

```release-note:none

```

### CODE REVIEW

File: mmv1/third_party/terraform/fwprovider/framework_provider_region_zone_test.go [hunk 1]
```diff
@@ -0,0 +1,395 @@
...
+	"testing"
+
+	"github.com/hashicorp/terraform-plugin-testing/helper/resource"
+	"github.com/hashicorp/terraform-provider-google/google/acctest"
+	"github.com/hashicorp/terraform-provider-google/google/envvar"
+)
+
+// TestAccFwProvider_region is a series of acc tests asserting how the plugin-framework provider handles credentials arguments
+// It is PF specific because the HCL used uses a PF-implemented data source
+// It is a counterpart to TestAccFwProvider_region
```
Comment (SarahFrench) [inline mmv1/third_party/terraform/fwprovider/framework_provider_region_zone_test.go:14]:
```suggestion
// It is a counterpart to TestAccSdkProvider_region
```

File: mmv1/third_party/terraform/fwprovider/framework_provider_region_zone_test.go [hunk 2]
```diff
@@ -0,0 +1,395 @@
...
+		tc := tc
+		t.Run(name, func(t *testing.T) {
+			tc(t)
+		})
+	}
+}
+
+// TestAccFwProvider_zone is a series of acc tests asserting how the plugin-framework provider handles credentials arguments
+// It is PF specific because the HCL used uses a PF-implemented data source
+// It is a counterpart to TestAccFwProvider_zone
```
Comment (SarahFrench) [inline mmv1/third_party/terraform/fwprovider/framework_provider_region_zone_test.go:37]:
```suggestion
// It is a counterpart to TestAccSdkProvider_zone
```

**Review by SarahFrench — COMMENTED** [review]
Comment fixes

**Comment by SarahFrench** [comment]
The tests in this PR are VCR-skipped, so here are some TC builds running the tests (FYI @BBBmau)
- SDK: https://hashicorp.teamcity.com/buildConfiguration/TerraformProviders_GoogleCloud_GOOGLE_MMUPSTREAMTESTS_GOOGLE_PACKAGE_PROVIDER/230372?hideTestsFromDependencies=false&hideProblemsFromDependencies=false&expandBuildChangesSection=true
- Framework: https://hashicorp.teamcity.com/buildConfiguration/TerraformProviders_GoogleCloud_GOOGLE_MMUPSTREAMTESTS_GOOGLE_PACKAGE_FWPROVIDER/230373?mode=branches&buildTab=overview&hideTestsFromDependencies=false&hideProblemsFromDependencies=false&expandBuildChangesSection=true

**Review by BBBmau — APPROVED** [review]
looks good! thanks for linking the teamcity tests!

**Comment by SarahFrench** [comment]
Thanks for the review!

**Pull request merged**